<launch>
  <arg name="max_vel" value="1.0" />
  <arg name="max_acc" value="0.5" />
  <arg name="dynamic" value="0" />
  <arg name="pcd_file" value="$(find prometheus_plan_manage)/config/cloud0115.pcd" />

  <!-- main node -->
  <node pkg="prometheus_plan_manage" name="dyn_planner_node" type="dyn_planner_node" output="screen">
  <!-- remap -->
    <!--remap from="/laser_cloud_surround" to="/random_forest/all_map"/-->
    <!--remap from="/odom_world" to="/state_ukf/odom"/-->

  <!-- sdf map updating -->
    <param name="sdf_map/SDF_MODE" value="1" type="int"/>
    <param name="sdf_map/origin_x" value="-8.0" type="double"/>
    <param name="sdf_map/origin_y" value="-7.0" type="double"/>
    <param name="sdf_map/origin_z" value="0.0" type="double"/>

    <param name="sdf_map/map_size_x" value="16.0" type="double"/>
    <param name="sdf_map/map_size_y" value="14.0" type="double"/>
    <param name="sdf_map/map_size_z" value="5.0" type="double"/>

    <param name="sdf_map/resolution_sdf" value="0.2" type="double"/>
    <param name="sdf_map/ceil_height" value="3.5" type="double"/>
    <param name="sdf_map/update_rate" value="2.0" type="double"/>
    <param name="sdf_map/update_range" value="8.0" type="double"/>
    <param name="sdf_map/inflate" value="0.1" type="double"/>
    <param name="sdf_map/radius_ignore" value="1.0" type="double"/>

  <!-- planning fsm -->
    <param name="fsm/flight_type" value="1" type="int"/>
    <param name="fsm/thresh_replan" value="2.0" type="double"/>
    <param name="fsm/thresh_no_replan" value="3.0" type="double"/>

    <param name="fsm/wp_num" value="3" type="int"/>

    <param name="fsm/wp0_x" value="-5.5" type="double"/>
    <param name="fsm/wp0_y" value="-4.0" type="double"/>
    <param name="fsm/wp0_z" value="1.0" type="double"/>

    <param name="fsm/wp1_x" value="0.0" type="double"/>
    <param name="fsm/wp1_y" value="-6.0" type="double"/>
    <param name="fsm/wp1_z" value="1.0" type="double"/>

    <param name="fsm/wp2_x" value="2.0" type="double"/>
    <param name="fsm/wp2_y" value="-3.0" type="double"/>
    <param name="fsm/wp2_z" value="0.5" type="double"/>

  <!-- planner manager -->
    <param name="manager/time_sample" value="0.5" type="double"/>
    <param name="manager/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="manager/dynamic" value="$(arg dynamic)" type="int"/>
    <param name="manager/margin" value="0.1" type="double"/>

  <!-- path searching -->
    <param name="astar/lambda_heu" value="10.0" type="double"/>
    <param name="astar/resolution_astar" value="0.1" type="double"/>
    <param name="astar/time_resolution" value="0.8" type="double"/>
    <param name="astar/margin" value="0.1" type="double"/>
    <param name="astar/allocate_num" value="100000" type="int"/>

  <!-- path searching -->
    <param name="search/max_tau" value="0.8" type="double"/>
    <param name="search/init_max_tau" value="0.8" type="double"/>
    <param name="search/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="search/max_acc" value="$(arg max_acc)" type="double"/>
    <param name="search/w_time" value="15.0" type="double"/>
    <param name="search/horizon" value="9.0" type="double"/>
    <param name="search/lambda_heu" value="5.0" type="double"/>
    <param name="search/resolution_astar" value="0.1" type="double"/>
    <param name="search/time_resolution" value="0.8" type="double"/>
    <param name="search/margin" value="0.4" type="double"/>
    <param name="search/allocate_num" value="100000" type="int"/>
    <param name="search/check_num" value="5" type="int"/>

  <!-- trajectory optimization -->
    <param name="optimization/lamda1" value="10.0" type="double"/>
    <param name="optimization/lamda2" value="0.6" type="double"/>
    <param name="optimization/lamda3" value="0.00001" type="double"/>
    <param name="optimization/lamda4" value="0.001" type="double"/>
    <param name="optimization/dist0" value="0.8" type="double"/>
    <param name="optimization/dist1" value="0.0" type="double"/>
    <param name="optimization/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="optimization/max_acc" value="$(arg max_acc)" type="double"/>

    <param name="optimization/algorithm" value="11" type="int"/>
    <param name="optimization/max_iteration_num" value="100" type="int"/>
    <param name="optimization/order" value="3" type="int"/>

    <param name="bspline/limit_vel" value="$(arg max_vel)" type="double"/>
    <param name="bspline/limit_acc" value="$(arg max_acc)" type="double"/>
    <param name="bspline/limit_ratio" value="1.1" type="double"/>
  </node>

  <!-- trajectory server -->
  <node pkg="prometheus_plan_manage" name="traj_server" type="traj_server" output="log">
      <remap from="/odom_world" to="/planning/odom_world"/>
      <remap from="/position_cmd" to="/planning/pos_cmd"/>
  </node>

    <!-- trajectory server -->
  <!-- <node pkg="prometheus_plan_manage" name="test_planning_static" type="test_planning_static" output="screen"> -->
      <!-- <param name="test/init_odom_x" value="1.0" type="double"/> -->
      <!-- <param name="test/init_odom_y" value="1.0" type="double"/> -->
      <!-- <param name="test/init_odom_z" value="1.0" type="double"/> -->
      <!-- <param name="test/file_name" value="$(arg pcd_file)" type="string"/> -->
      <!-- <param name="test/odom_mode" value="1" type="int"/> -->
  <!-- </node> -->



</launch>